cli/tests/test_upgrade_release_smoke_contract.py

```
@@ -251,6 +256,157 @@ def test_upgrade_failure_guidance_does_not_restore_gateway_jsonl_ownership() ->
+    start = source.index(start_marker)
+    end = source.index(end_marker, start) + len(end_marker)
+    return source[start:end] + "\n"
+
+
+def _run_bridge_comment_restore(
+    source_path: Path,
+    active_path: Path,
+    *,
+    program: str | None = None,
+) -> subprocess.CompletedProcess[str]:
+    return subprocess.run(
```

Ruff (0.15.21)

[error] `subprocess` call: check for execution of untrusted input

(S603)